test.py

```diff
@@ -107,10 +107,10 @@ def download_test(self, filename, recursive, destination=None,
         os.mkdir(temp_in)
         os.chdir(temp_in)
         try:
-            with SCPClient(self.ssh.get_transport()) as scp:
-                scp.get(filename,
-                        destination if destination is not None else u'.',
-                        preserve_times=True, recursive=recursive)
+            get(self.ssh.get_transport(),
```

Comment:
Do the tests take long to run? Otherwise it might be worthwhile to test `SCPClient` as well. You're now only testing half of the public API.

I realize that put/get just delegate to SCPClient put/get currently, but it seems bad for the tests to assume that.